### PR TITLE
Properly handle out of voltage bus error

### PR DESCRIPTION
**Pull request by ne0ds**

### CODE REVIEW

File: src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitWorkerService.java [hunk 1]
```diff
@@ -143,19 +143,23 @@ private List<Fault> getAllBusfaultFromNetwork(Network network, ShortCircuitRunCo
...
         String busId = context.getBusId();
         Identifiable<?> identifiable = network.getIdentifiable(busId);
         Map<String, ShortCircuitLimits> shortCircuitLimits = new HashMap<>();
 
-        if (identifiable instanceof BusbarSection) {
-            String busIdFromBusView = ((BusbarSection) identifiable).getTerminal().getBusView().getBus().getId();
+        if (identifiable instanceof BusbarSection busbarSection) {
+            Bus bus = busbarSection.getTerminal().getBusView().getBus();
+            if (bus == null) {
+                notificationService.publishFail(resultUuid, context.getReceiver(), "Selected bus is out of voltage", context.getUserId(), busId);
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitWorkerService.java:154]: You can replace those two rows by something like : 
`throw new PowsyblException("Selected bus" + busId + " is out of voltage");`

So you're homogeneous with row 173, you don't need to give resultUuid as parameter and you can remove the change at row 191

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java [hunk 2]
```diff
@@ -67,31 +115,46 @@ void testLogsMappersIsCalled() throws Exception {
...
+        var busId = "bus1";
+
+        when(context.getBusId()).thenReturn(busId);
+        doReturn(busbarSection).when(network).getIdentifiable(busId);
+        when(busbarSection.getTerminal()).thenReturn(terminal);
+        when(terminal.getBusView()).thenReturn(busView);
+        when(busView.getBus()).thenReturn(null);
+
+        var exception = (ShortCircuitException) catchRuntimeException(() -> workerService.getBusFaultFromBusId(network, context));
+        assertThat(exception.getType()).isEqualTo(BUS_OUT_OF_VOLTAGE);
```
Comment (klesaulnier) [inline src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java:154]: @etiennehomer 
I'm not sure we should use "var", what do you think ?
In such a small scope, I guess it's ok

File: src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java [hunk 3]
```diff
@@ -63,35 +118,54 @@ void testLogsMappersIsCalled() throws Exception {
...
+            workerService.consumeRun().accept(message);
+        } catch (ShortCircuitException expectedException) {
+            assertThat(expectedException.getType()).isEqualTo(BUS_OUT_OF_VOLTAGE);
         }
     }
 
-    @AllArgsConstructor
-    private static class ShortCircuitAnalysisProviderMock implements ShortCircuitAnalysisProvider {
-        private final ShortCircuitAnalysisResult result;
+    private record ShortCircuitAnalysisProviderMock(ShortCircuitAnalysisResult result) implements ShortCircuitAnalysisProvider {
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java:168]: Not to be done in this PR. And i'm not sure it's relevant to use record here

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java (hunk 2 above)
Comment (etiennehomer) [reply src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java]: Yes why not in a small test method like this

File: src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java [hunk 4]
```diff
@@ -63,29 +118,52 @@ void testLogsMappersIsCalled() throws Exception {
...
+
+        when(runContext.getBusId()).thenReturn(busId);
+        when(networkStoreService.getNetwork(any(), any())).thenReturn(network);
+        doReturn(busbarSection).when(network).getIdentifiable(busId);
+        when(network.getVariantManager()).thenReturn(variantManager);
+        when(busbarSection.getTerminal()).thenReturn(terminal);
+        when(terminal.getBusView()).thenReturn(busView);
+        when(busView.getBus()).thenReturn(null);
+
+        try (var shortCircuitAnalysisMockedStatic = TestUtils.injectShortCircuitAnalysisProvider(analysisProvider);
```
Comment (etiennehomer) [inline src/test/java/org/gridsuite/shortcircuit/server/service/ShortCircuitServiceTest.java:162]: shortCircuitAnalysisMockedStatic is not used.

**Review by etiennehomer — CHANGES_REQUESTED** [review]

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**